controllers/backstage_route.go

```diff
@@ -110,3 +115,13 @@ func (r *BackstageReconciler) applyRouteParamsFromCR(route *openshift.Route, bac
 		}
 	}
 }
+
+func shouldCreateRoute(backstage bs.Backstage) bool {
+	if backstage.Spec.Application == nil {
```

Comment:
This can be simplified:
return backstage.Spec.Application == nil || backstage.Spec.Application.Route == nil || pointer.BoolDeref(backstage.Spec.Application.Route.Enabled, true)

Reply:
Yes, I tried to favor readability, by not having a too long condition.